src/hooks/useAddCasesInputs.tsx

```diff
@@ -17,6 +17,31 @@ export const getBracketData = (modelInputs: ModelInputs) => {
   return pick(modelInputs, populationBracketKeys);
 };
 
+function findMostRecentDate(
+  observedAtDate: Date,
+  facilityModelVersions: ModelInputs[] | undefined,
+) {
+  let mostRecentDate = observedAtDate;
+  if (facilityModelVersions) {
+    // create array of dates with observed data at a given facility
+    const facilityObservedAtDates = facilityModelVersions.map(
+      (facility) => facility.observedAt,
+    );
+    // filter to dates earlier than (or the same as) the current date
+    const earlierDates = facilityObservedAtDates?.filter(function (date) {
+      return startOfDay(date) <= startOfDay(observedAtDate);
+    });
+    // if there is data for prior dates, use the most recent one, otherwise use
+    // the next forward-looking date that we have for the facility
+    if (earlierDates && earlierDates.length > 0) {
+      mostRecentDate = earlierDates[earlierDates.length - 1];
```

Comment:
The indexing logic in this conditional block assumes that `facilityModelVersions` has already been sorted by `observedAt`, right? Just making sure that that's a safe assumption?

Reply:
@jessex from my local testing it looks like it's already sorted, but I'll add an explicit sort to make it more robust

Reply:
@macfarlandian to confirm that's a good call and not superfluous. Ian, do you think an extra sort here is called for?

Reply:
it should generally be a safe assumption that the dates are already sorted, so I don't think it's really necessary here